src/Symfony/Component/Console/Input/InputInterface.php

```diff
@@ -107,6 +107,13 @@ public function hasArgument(string $name): bool;
      */
     public function getOptions(): array;
 
+    /**
+     * Returns all the given options without adding any default values.
+     *
+     * @return array<string|bool|int|float|array|null>
+     */
+    public function getNonDefaultOptions(): array;
```

Comment:
Adding a new method in an interface is a BC break so it cannot be done. It needs to be done as a forward compatibility layer instead.

Reply:
Where do you suggest to define it ?

Reply:
you need to define the method using `@method` in the interface instead of actually adding the method, to tell the tooling that the method will be added in the interface in the next major version.